icu4c/source/common/uresbund.cpp

```diff
@@ -2350,7 +2350,84 @@ struct GetAllChildrenSink : public ResourceSink {
+                                                                             stackTempBundle.getAlias(), &errorCode);
+                                    tablePath.clear();
+                                    tablePath.append(aliasRB->fResPath, errorCode);
+                                    entry = aliasRB->fData;
+                                    res = aliasRB->fRes;
+                                    aliasedValue.setData(entry->fData);
+                                    // do I also need to call aliasedValue.setValueLocaleDataEntry() ?
+                                    aliasedValue.setResource(res, ResourceTracer(aliasRB)); // probably wrong to use aliasRB here
+                                    aliasedValueType = aliasedValue.getType();
+                                }
+                                if (aliasedValueType == URES_TABLE) {
+                                    dest.put(parentKey, aliasedValue, isRoot, errorCode);
+                                }
+                            }
+                        }
+
+                            
+//                            ICUResourceBundle newAliasedResource = aliasedResource.getParent().findWithFallback(tablePath);
+////                            if (newAliasedResource.key.equals(aliasedResource.key)) {
+//                            aliasedResource = newAliasedResource;
+////                            } else {
+////                                // the findWithFallback() call above might follow an alias.  If it does, we'll get
+////                                // back the alias target at the wrong level (e.g., if we're in en_CA, we're calling
+////                                // findWithFallback() on en, and if it follows an alias, we get back the alias target
+////                                // in en, even if it also exists in en_CA).  So we check the keys to see if we followed
+////                                // an alias, and if we did, we re-fetch the alias target from our original resource bundle
+////                                tablePath = tablePath.substring(0, tablePath.lastIndexOf('/'));
+////                                tablePath += "/" + newAliasedResource.key;
+////                                aliasedResource = ICUResourceBundle.this.findWithFallback(tablePath);
+////                            }
+//                            aliasedResourceImpl = (ICUResourceBundleImpl) aliasedResource;
+//                            aliasedValue = new ReaderValue();
+//                            aliasedValue.reader = aliasedResourceImpl.wholeBundle.reader;
+//                            aliasedValue.res = aliasedResourceImpl.getResource();
+//                            aliasedValueType = aliasedValue.getType(); // sink.put() messes up the value
+//                            keyCopy = key.clone(); // sink.put() messes up the key
+//                            sink.put(keyCopy, aliasedValue, noFallback);
+//                        }
```

Comment:
I'm sorry.  I'll take out all this crap in a future commit after I've received review feedback.